File: BackEnd/Backend_Java/SoftPacModel/src/main/java/pe/edu/pucp/softpac/model/CuentasPropiasDTO.java
package pe.edu.pucp.softpac.model;

import java.math.BigDecimal;

public class CuentasPropiasDTO extends CuentasBancariasDTO{
    private BigDecimal saldo_disponible;
    public CuentasPropiasDTO(){
        super();
        saldo_disponible=new BigDecimal("0.0");
    }
    
    public CuentasPropiasDTO(Integer cuenta_bancaria_id, String tipo_cuenta, String cci, Boolean activa, String numero_cuenta, MonedasDTO moneda, EntidadesBancariasDTO entidad_bancaria, BigDecimal saldo_disponible){
        super(cuenta_bancaria_id,tipo_cuenta,cci,activa,numero_cuenta,moneda,entidad_bancaria);
        this.saldo_disponible=saldo_disponible;
    }

    public CuentasPropiasDTO(CuentasPropiasDTO cuentaPropia){
        super(cuentaPropia);
        this.saldo_disponible=cuentaPropia.getSaldo_disponible();
    }
    /**
     * @return the saldo_disponible
     */
    public BigDecimal getSaldo_disponible() {
        return saldo_disponible;
    }

    /**
     * @param saldo_disponible the saldo_disponible to set
     */
    public void setSaldo_disponible(BigDecimal saldo_disponible) {
        this.saldo_disponible = saldo_disponible;
    }
}

File: FrontEnd/SoftPac/SoftPacModel/CuentasPropiasDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPac.Model
{
    [Serializable]
    public class CuentasPropiasDTO : CuentasBancariasDTO
    {
        private decimal saldoDisponible;

        public CuentasPropiasDTO() : base()
        {
            SaldoDisponible = 0.0M;
        }

        public CuentasPropiasDTO(int cuentaBancariaId, string tipoCuenta, string cci, bool activa,
                                  string numeroCuenta, MonedasDTO moneda, EntidadesBancariasDTO entidadBancaria,
                                  decimal saldoDisponible)
            : base(cuentaBancariaId, tipoCuenta, cci, activa, numeroCuenta, moneda, entidadBancaria)
        {
            SaldoDisponible = saldoDisponible;
        }

        public CuentasPropiasDTO(CuentasPropiasDTO other) : base(other)
        {
            this.SaldoDisponible = other.SaldoDisponible;
        }

        public decimal SaldoDisponible { get => saldoDisponible; set => saldoDisponible = value; }

    }
}
